docs/Branches.md

```diff
@@ -0,0 +1,73 @@
+
+For some releases, the release branch for the LLVM projects (LLVM, Clang, compiler-rt, and LLDB) will be based on the corresponding llvm.org release branch.
+
+
+## The Upstream Branches
+
+| Swift       | LLVM                | LLDB 
+| ----------- | ------------------- | -------------------
+| master-next | upstream-with-swift | upstream-with-swift
+
+`upstream-with-swift` is a branch for LLVM that includes all changes necessary to support Swift. Changes from llvm.org's master branch are automatically merged in. Why isn't this just `stable`? Well, because LLVM changes *very* rapidly, and that wouldn't be very stable. However, release branches for LLVM and Clang *do* include the latest upstream changes, and so they come from this branch. That means they need to include any necessary Swift support.
+
+If you are making changes to LLVM to support Swift, you'll probably need to work on them in `stable` to test them against Swift itself, but they should be committed to `upstream-with-swift`, and possibly cherry-picked to the current release branch (`swift-x.y-branch`) if needed.
+
+(If you're making changes to LLVM or LLDB that *aren't* about Swift, they should generally be made on llvm.org instead.)
+
+`master-next` is an effort to keep Swift building with the latest LLVM changes. Ideally when LLVM changes, no Swift updates are needed, but that isn't always the case. In these situations, any adjustments can go into Swift's `master-next` branch. Changes from Swift's `master` are automatically merged into `master-next` as well. The `stable-next` branch represents a known good state from the `upstream-with-swift` branch that `master-next` can build against. It should never have any extra commits.
+
+LLDB's `upstream-with-swift` has *both* aspects: changes are automatically merged in from `stable` *and* from llvm.org's master branch. Again, ideally there are no changes necessary here, but in practice LLDB may need updates to continue building against LLVM's `upstream-with-swift` and Swift's `master-next`.
+
+# Reference
+
+## Committing
+
+- Swift: new commits go to `master`, then cherry-pick to the release branch (`swift-x.y-branch`) if necessary.
+
+- LLVM/Clang/compiler-rt: new commits go to `upstream-with-swift`, then cherry-pick to the release branch (`swift-x.y-branch`) if necessary.
+
+- LLDB: new commits go to the release branch (`swift-x.y-branch`) unless not meant for the current release. (In that case, use `stable`.)
```

Comment:
Updating based on overheard discussion with @jimingham today. Better to just start with `stable` and cherry-pick, like we do with Swift.